process_report/tests/e2e/test_e2e_pipeline.py

```diff
@@ -178,24 +181,35 @@ def _validate_outputs(workspace: Path) -> None:
         AssertionError: If any expected output file is missing, empty, or invalid.
         pytest.fail: If CSV files cannot be read or parsed.
     """
-    logger = logging.getLogger(__name__)
     logger.info(f"Validating pipeline outputs in: {workspace}")
 
-    # Validate CSV files
+    # Basic existence and structure validation catches pipeline execution failures
     for csv_file in EXPECTED_CSV_FILES:
         csv_path = workspace / csv_file
         assert csv_path.exists(), f"CSV file not found: {csv_path}"
         assert csv_path.is_file(), f"Path is not a file: {csv_path}"
         assert csv_path.stat().st_size > 0, f"CSV file is empty: {csv_path}"
 
-        # Check file can be read as CSV
+        # Basic parsing check ensures files aren't corrupted or malformed
         try:
             df = pd.read_csv(csv_path)
             assert len(df.columns) > 0, f"CSV has no columns: {csv_path}"
         except Exception as e:
             pytest.fail(f"Failed to read CSV {csv_path}: {e}")
 
-    # Validate PI invoices directory
+    # Validate that only expected CSV files are created (no more, no less)
+    actual_csv_files = set(csv_path.name for csv_path in workspace.glob("*.csv"))
+    expected_csv_files = set(EXPECTED_CSV_FILES)
+
+    unexpected_files = actual_csv_files - expected_csv_files
+    if unexpected_files:
+        pytest.fail(f"Unexpected CSV files created: {sorted(unexpected_files)}")
+
+    missing_files = expected_csv_files - actual_csv_files
+    if missing_files:
+        pytest.fail(f"Expected CSV files missing: {sorted(missing_files)}")
```

Comment:
This could be collasped into one test:
```suggestion
assert not (actual_csv_files - expected_csv_files), f"Output dir contains unexpected or missing CSVs: {actual_csv_files}"
```

Reply:
@naved001 I was thinking that which files were missing/unexpected could be implied from inspecting `actual_csv_files`, but I guess having it as it is is fine as well.